admin server init commit

neondude opened the pull request:
Admin server for authentication and other CEP related operations such push notifications, etc.

neondude committed:
admin server init commit

SuryaprakashCIT commented on admin_server/routes/auth.js:
```diff
@@ -0,0 +1,146 @@
+const axios = require('axios');
+const jwt = require('jsonwebtoken');
+
+const MSG91_URL = "https://control.msg91.com/api/";
```
Can we have this MSG91_URL as an ENV ... will be easy if we had to change it tomorrow.

neondude replied:
if we change service, endpoint will not be the same. even if we create a wrapper function the params will not be the same.

SuryaprakashCIT replied:
I mentioned the URL string.

SuryaprakashCIT commented on admin_server/routes/auth.js:
```diff
@@ -0,0 +1,146 @@
+    res.status(500).send({
+      message: "internal error",
+      status: "failed",
+      error: ""+error
```
Do we need to send the error in the response?. Logging the error might be good, we dont know the size of the error trace.

neondude replied:
only the error message should be fine. it can be useful for debugging

SuryaprakashCIT replied:
What i meant was ... status code 500 corresponds to internal server error. Do we need to specify that explicitly ??.
Also we are appending the error object in line 44.

neondude replied:
for the sake of consistency and ease of development, i would like to have a consistent response message for every call. 
In line 44 since i have appended with empty string (""+) only error message will be printed, not the entire object

SuryaprakashCIT commented on admin_server/routes/auth.js:
```diff
@@ -0,0 +1,146 @@
+      res.status(200).send({
+        message: "sms sent",
+        status: "success"
+      });
```
Think the status codes are self explanatory. do we need a seperate status message be sent back in the response ?

neondude replied:
for the sake of consistency and ease of development, i would like to have a consistent response message for every call. In any case even in regards to failure when implementing the api, the developer can expect a consistent response.

SuryaprakashCIT commented on admin_server/routes/auth.js:
```diff
@@ -0,0 +1,146 @@
+            "x-hasura-allowed-roles": ["user"],
+            "x-hasura-default-role": "user",
+            'X-Hasura-Mobile-No': req.body.mobile_no
+          },
```
So the idea is to directly use these data as request headers ... right ?

neondude replied:
the client has to add the given token to the  header Authorization : Bearer <token>

SuryaprakashCIT replied:
Cool

neondude committed:
changes hasura claim value tostring

hasura expects all values to be string

neondude committed:
change hasura claims key to lower case

hasura token claims keys are case sensitive. changed to lower case for
simplicity
